sdks/python/apache_beam/runners/dataflow/internal/apiclient.py

```diff
@@ -512,7 +512,8 @@ def create_job(self, job):
     if job_location:
       gcs_or_local_path = os.path.dirname(job_location)
       file_name = os.path.basename(job_location)
-      self.stage_file(gcs_or_local_path, file_name, io.BytesIO(job.json()))
+      self.stage_file(gcs_or_local_path, file_name,
+                      io.BytesIO(job.json().encode('utf-8')))
```

Comment:
Pipeline works well without this change. Wondering what problem you want to fix here?

Reply:
`io.BytesIO()` expects `bytes`, while `job.json()` returns a `string`. Is this code path executed when running your pipeline?

Reply:
I don't think we hit this codepath before, the change makes sense to me.